### DISCUSSION TITLE
working example of ssh asset in Dagster

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18475] [Q&A] [answered]
Does anyone has a concise and clear working example of an ssh ~task~ asset in Dagster? It can be as simple as doing a `touch /tmp/helloworld` on a remote host.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1701470239631849?thread_ts=1701470239.631849&cid=C01U954MEER).*

**Comment by yuhan** [comment, marked as answer]
```python

from dagster import asset
from dagster_ssh import SSHResource


@asset
def my_asset(ssh: SSHResource):
    ssh.sftp_get("/path/to/remote.csv", "path/to/local.csv")


defs = Definitions(
    assets=[my_asset],
    resources={
        "ssh": SSHResource(
            remote_host="foo.com",
            key_file="path/to/id_rsa",
        ),
    },
)

```
